### PR TITLE
Fix bad cudagraph interaction

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104196

Fix for https://github.com/pytorch/pytorch/issues/103126

As mentioned there, 

> We need to make sure we are not removing the misaligned inputs before we are checking for misalignment in cudagraphs, so we know not to expect a static input for the misaligned tensors.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78

### CODE REVIEW

File: torch/_inductor/cudagraph_trees.py [hunk 1]
```diff
@@ -1506,6 +1515,10 @@ def check_memory_pool(device, pool_id, live_storages_ptrs: List[StorageWeakRefWr
     if torch._C._cuda_checkPoolLiveAllocations(device, pool_id, unique_storages):
         return
 
+    # at this point we are past the fast-path. we have seen rare cases where a dead tensor is dead,
```
Comment (eellison) [inline torch/_inductor/cudagraph_trees.py:1518]: this is unrelated but sneaking it in since it's a 1-line change

**Comment by eellison** [comment]
test failure unrelated

File: torch/_inductor/compile_fx.py [hunk 2]
```diff
@@ -372,7 +372,13 @@ def compiled_artifact(new_inputs):
                         "skipping cudagraphs due to multiple device indexes"
                     )
 
-    result = align_inputs(compiled_graph, example_inputs, range(num_fixed))
+    # cudagraphs does its own aligning of inputs
+    if not cudagraphs:
+        new_callable = align_inputs(
```
Comment (desertfire) [inline torch/_inductor/compile_fx.py:377]: Nit: we could always assign to `compiled_graph.current_callable` without doing the checking next.

**Review by desertfire — APPROVED** [review]

File: torch/_inductor/compile_fx.py (hunk 2 above)
Comment (eellison) [reply torch/_inductor/compile_fx.py:377]: true, will add on in follow up since i have tests passing..

**Pull request closed without merging**